react-main/src/app/components/Game/index.js

```diff
@@ -2,53 +2,67 @@ import React, { Component } from 'react';
 
 import Board from '../Board';
 
+import { connect } from 'react-redux';
+import { selectSquare, selectHistory, getWinner } from '../../../redux/game/actions';
+
 import styles from './styles.scss';
-import calculateWinner from './utils.js';
 
 class Game extends Component {
-  state = { history: [{ squares: Array(9).fill(null) }], xIsNext: true, stepNumber: 0 };
-
-  jumpTo = move => this.setState({ stepNumber: move, xIsNext: move % 2 === 0 });
-
-  handleClick = i => {
-    const history = this.state.history;
-    const squares = history[this.state.stepNumber].squares.slice();
-    if (calculateWinner(squares) || squares[i]) {
-      return;
-    }
-
-    squares[i] = this.state.xIsNext ? 'X' : 'O';
-    this.setState({
-      history: history.concat([{ squares }]),
-      xIsNext: !this.state.xIsNext,
-      stepNumber: this.state.stepNumber + 1
-    });
+
+  handleSquareClick = i => (
+      this.props.square(i)
+  );
+
+  handleHistoryClick = step => {
+      return this.props.historySelected(step);
+  };
+
+  componentDidMount(){
+     this.props.winner()
+  };
+
+  componentWillUpdate(){
```

Comment:
space before braces

Reply:
Also, avoid componentWillUpdate. This method will be deprecated in the next react version. Use componentDidUpdate for this